includes/class-sensei-course.php

```diff
@@ -145,17 +145,87 @@ public function __construct() {
+		add_action( 'submenu_file', [ $this, 'highlight_menu_item' ] );
+	}
+
+	/**
+	 * Highlight the menu item for the course pages.
+	 *
+	 * @since 4.0.0
+	 * @access private
+	 *
+	 * @param string $submenu_file
+	 *
+	 * @return string
+	 */
+	public function highlight_menu_item( $submenu_file ) {
+		$screen = get_current_screen();
+
+		if ( $screen && in_array( $screen->id, [ 'edit-course', 'edit-course-category', 'course_page_course-order' ], true ) ) {
+			$submenu_file = 'edit.php?post_type=course';
+		}
+
+		return $submenu_file;
 	}
 
 	/**
-	 * Register and enqueue scripts that are needed in the backend.
+	 * Add custom navigation to the admin pages.
+	 *
+	 * @since 4.0.0
+	 * @access private
+	 */
+	public function add_custom_navigation() {
+		$screen = get_current_screen();
+		if ( ! $screen ) {
+			return;
+		}
+		if ( in_array( $screen->id, [ 'edit-course', 'edit-course-category' ], true ) ) {
+			$this->display_courses_navigation( $screen );
+		}
+	}
+
+	/**
+	 * Display the courses' navigation.
+	 *
+	 * @param WP_Screen $screen
+	 */
+	private function display_courses_navigation( WP_Screen $screen ) {
+		?>
+		<div id="sensei-custom-navigation" class="sensei-custom-navigation">
+			<div class="sensei-custom-navigation__main">
+				<div class="sensei-custom-navigation__main__title">
+					<h1><?php esc_html_e( 'Courses', 'sensei-lms' ); ?></h1>
+				</div>
+				<div class="sensei-custom-navigation__main__tabbar">
+					<a class="sensei-custom-navigation__main__tabbar__tab <?php echo '' === $screen->taxonomy ? 'active' : ''; ?>" href="<?php echo esc_url( admin_url( 'edit.php?post_type=course' ) ); ?>"><?php esc_html_e( 'All Courses', 'sensei-lms' ); ?></a>
+					<a class="sensei-custom-navigation__main__tabbar__tab <?php echo 'course-category' === $screen->taxonomy ? 'active' : ''; ?>" href="<?php echo esc_url( admin_url( 'edit-tags.php?taxonomy=course-category&post_type=course' ) ); ?>"><?php esc_html_e( 'Course Categories', 'sensei-lms' ); ?></a>
+				</div>
+			</div>
+			<div class="sensei-custom-navigation__additional">
+				<a class="page-title-action" href="<?php echo esc_url( admin_url( 'post-new.php?post_type=course' ) ); ?>"><?php esc_html_e( 'New course', 'sensei-lms' ); ?></a>
+				<a href="<?php echo esc_url( admin_url( 'edit.php?post_type=course&page=course-order' ) ); ?>"><?php esc_html_e( 'Order courses', 'sensei-lms' ); ?></a>
+				<a href="<?php echo esc_url( admin_url( 'edit.php?post_type=course&page=sensei-settings#course-settings' ) ); ?>"><?php esc_html_e( 'Course settings', 'sensei-lms' ); ?></a>
+			</div>
+		</div>
+		<?php
+	}
+
+	/**
+	 * Register and enqueue scripts and styles that are needed in the backend.
 	 *
 	 * @access private
 	 * @since 2.1.0
 	 */
 	public function register_admin_scripts() {
 		$screen = get_current_screen();
+		if ( ! $screen ) {
+			return;
+		}
 
+		Sensei()->assets->enqueue( 'sensei-admin-custom-navigation', 'js/admin/custom-navigation.js', [], true );
```

Comment:
This script is loading on quite a lot of other pages where it's not needed (e.g.  Modules, Lessons etc.). Could we change to loading it only on pages where the header appears?

Reply:
The script is universal and should work for further implementations of the navigation in Modules, Lessons etc.
But, probably, I should to move this statement to the `Sensei_Admin` class. 
WDYT?

Reply:
Yeah, probably makes more sensei to put it in `Sensei_Admin`. Regardless of where it lives though, we'd still need to introduce a few checks for the course-related pages, because even though we ultimately need it to be available on other pages like _Modules_ (although subsequent PRs would be the most appropriate place to add more checks for more pages), it still loads for pages like _Grading_, _Student Management_ etc.

Reply:
Fixed here: https://github.com/Automattic/sensei/pull/4672/commits/c3df7fb99da39851bf561bff7a8a8621cd7b810a